test/parallel/test-fs-realpath-pipe.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+// Skip Windows as /dev/stdin is not available
+if (process.platform === 'win32')
+  process.exit();
+
+const common = require('../common');
+const assert = require('assert');
+
+const expected = '/dev/stdin';
+
+const spawn = require('child_process').spawn;
+
+for (const code of [
+  `require('fs').realpath('/dev/stdin', (err, resolvedPath) => {
+    console.log(resolvedPath);
+  });`,
+  `console.log(require('fs').realpathSync('/dev/stdin'));`
+]) {
+  const child = spawn(process.execPath, ['-e', code], { stdio: 'pipe' });
```

Comment:
Consider using [`spawnSync()`](https://nodejs.org/docs/latest/api/child_process.html#child_process_child_process_spawnsync_command_args_options) to simplify this a bit and to make future debugging a bit easier.